src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -3007,11 +3007,15 @@ public static OperationStatus FromHexString(ReadOnlySpan<char> source, Span<byte
                 quotient = destination.Length;
                 result = OperationStatus.DestinationTooSmall;
             }
-            else if (remainder == 1)
+            else if (destination.Length > quotient)
```

Comment:
Shouldn't we also be handling remainders in cases where the destination buffer precisely matches the quotient?
```suggestion
            else
```